<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Peer Assignment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a SQLIte database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. Click on the links below to download and save the datasets (.CSV files):

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Census Data</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Public Schools</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Crime Data</a>

**NOTE:** Ensure you have downloaded the datasets using the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**

Let us now load the ipython-sql  extension and establish a connection with the database

* Here you will be loading the csv files into the pandas Dataframe and then loading the data into the above mentioned sqlite tables.

* Next you will be connecting to the sqlite database  **FinalDB**.

Refer to the previous lab for hints .

<a href ="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/Module%205/DB0201EN-Week3-1-4-Analyzing_SQLite.ipynb">Hands-on Lab: Analyzing a real World Data Set</a>





In [1]:
%reload_ext sql

In [2]:
%%javascript 
console.log('Hello World')

<IPython.core.display.Javascript object>

In [3]:
#!pip install -q pandas==1.1.5

In [4]:
import csv, sqlite3, pandas as pd
con=sqlite3.connect('Chicago.db')
cursor_=con.cursor()

In [5]:
%sql sqlite:///Chicago.db


In [6]:
%cd ..
# %cd ..

C:\Users\AMOS OSHIYOYE\Documents\VSCODE_user_files\Coursera\Coursera


In [83]:
crime_data=pd.read_csv('../Coursera/CSVs/ChicagoCrimeData.csv')
census=pd.read_csv('../Coursera/CSVs/ChicagoCensusData.csv')
public_sch=pd.read_csv('../Coursera/CSVs/ChicagoPublicSchools.csv')

In [84]:
census.to_sql('census',con, if_exists='replace')
crime_data.to_sql('crime_data',con,if_exists='replace')
public_sch.to_sql('school_data',con, if_exists='replace')

566

## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [85]:
%sql Select count(*) as 'total number of rows' from crime_data

 * sqlite:///Chicago.db
Done.


[(533,)]

### Problem 2

##### List community areas with per capita income less than 11000.


In [190]:
%sql select community_area_name from census where per_capita_income<11000

 * sqlite:///Chicago.db
Done.


[('West Garfield Park',),
 ('South Lawndale',),
 ('Fuller Park',),
 ('Riverdale',)]

### Problem 3

##### List all case numbers for crimes  involving minors?(children are not considered minors for the purposes of crime analysis)


In [193]:
%sql select case_number from crime_data where DESCRIPTION like '%minor%'

 * sqlite:///Chicago.db
Done.


[('HL266884',), ('HK238408',)]

### Problem 4

##### List all kidnapping crimes involving a child?


In [88]:
%sql select * from crime_data where primary_type like '%kidnap%' and description like '%child%'

 * sqlite:///Chicago.db
Done.


[(520, 5276766, 'HN144152', '2007-01-26', '050XX W VAN BUREN ST', '1792', 'KIDNAPPING', 'CHILD ABDUCTION/STRANGER', 'STREET', 0, 0, 1533, 15, 29.0, 25.0, '20', 1143050.0, 1897546.0, 2007, 41.87490841, -87.75024931, '(41.874908413, -87.750249307)')]

### Problem 5

##### What kinds of crimes were recorded at schools?


In [194]:
%sql select primary_type,description from crime_data where location_description like '%school%'

 * sqlite:///Chicago.db
Done.


[('BATTERY', 'SIMPLE'),
 ('BATTERY', 'PRO EMP HANDS NO/MIN INJURY'),
 ('BATTERY', 'SIMPLE'),
 ('BATTERY', 'SIMPLE'),
 ('BATTERY', 'SIMPLE'),
 ('CRIMINAL DAMAGE', 'TO VEHICLE'),
 ('NARCOTICS', 'POSS: HEROIN(WHITE)'),
 ('NARCOTICS', 'MANU/DEL:CANNABIS 10GM OR LESS'),
 ('ASSAULT', 'PRO EMP HANDS NO/MIN INJURY'),
 ('CRIMINAL TRESPASS', 'TO LAND'),
 ('PUBLIC PEACE VIOLATION', 'BOMB THREAT'),
 ('PUBLIC PEACE VIOLATION', 'BOMB THREAT')]

### Problem 6

##### List the average safety score for each type of school.


In [61]:
%sql select `Elementary, Middle, or High School` as school_type,AVG(safety_score) as Average_safety_score from school_data group by `Elementary, Middle, or High School`

 * sqlite:///Chicago.db
Done.


[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [114]:
%sql select community_area_name,percent_households_below_poverty from census\
Order by percent_households_below_poverty desc\
limit 5


 * sqlite:///Chicago.db
Done.


[('Riverdale', 56.5),
 ('Fuller Park', 51.2),
 ('Englewood', 46.6),
 ('North Lawndale', 43.1),
 ('East Garfield Park', 42.4)]

### Problem 8

##### Which community area is most crime prone?


In [115]:
%sql select census.community_area_name, count(crime_data.community_area_number) \
as count_value from census,crime_data  on census.community_area_number=crime_data.community_area_number\
group by census.community_area_number\
order by count_value desc limit 1

 * sqlite:///Chicago.db
Done.


[('Austin', 43)]

Double-click **here** for a hint

<!--
Query for the 'community area number' that is most crime prone.
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [122]:
%sql select COMMUNITY_AREA_NAME from census where hardship_index =(select max(hardship_index) from census)

 * sqlite:///Chicago.db
Done.


[('Riverdale',)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [163]:
%sql select community_area_name as community_name,count(community_area_number) as count_value \
from (select census.community_area_name, crime_data.community_area_number\
from census,crime_data on census.community_area_number=crime_data.community_area_number) group by community_name\
order by count_value desc limit 1

 * sqlite:///Chicago.db
Done.


[('Austin', 43)]

In [185]:
# %sql select count(*) from school_data limit 1
%sql select * from school_data where `10th Grade PLAN (2010)` not like '%NDA%' limit 5

 * sqlite:///Chicago.db
Done.


[(4, 610513, 'Air Force Academy High School', 'HS', '3630 S Wells St', 'Chicago', 'IL', 60609, '(773) 535-1590', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610513.pdf', 'Southwest Side High School Network', 'SOUTHWEST SIDE COLLABORATIVE', 'NDA', 'Standard', 'Not on Probation', 'Not Enough Data', 'Yes', 'Average', 49.0, 'Strong', '60', 'Strong', 60.0, 'Average', 55.0, 'Weak', '45', 'Average', '54', 'Average', '53', 'Average', '49', '93.30%', 15.6, '96.90%', '100.00%', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', None, None, None, None, 'NDA', 'NDA', 'NDA', 'NDA', '14.6', '14.8', 'NDA', '16', '1.4', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 302, 40, '91.8', 1175177.622, 1880745.126, 41.82814609, -87.63279369, 34, 'ARMOUR SQUARE', 11, 9, '(41.82814609, -87.63279369)'),
 (6, 609720, 'Albert G Lane Technical High School', 'HS', '2501 W Addison St', 'Chicago', 'IL', 60618, '(773) 534-5400', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_609720.pdf', 'North-Northwest Side High School Network', 'NORTH-NORTHWEST SIDE COLLABORATIVE', 'Yes', 'Standard', 'Not on Probation', 'Level 1', 'No', 'Very Strong', 88.0, 'NDA', 'NDA', 'Strong', 62.0, 'Average', 52.0, 'Weak', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', '96.30%', 2.1, '96.20%', '99.40%', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', None, None, None, None, 'NDA', 'NDA', 'NDA', 'NDA', '19.1', '19.5', '19.9', '20.1', '1', '23.4', '3.5', '67.9', '92.2', '79.8', 4368, 35, '90.7', 1158975.392, 1923791.705, 41.94661693, -87.69105603, 5, 'NORTH CENTER', 47, 19, '(41.94661693, -87.69105603)'),
 (8, 610524, 'Alcott High School for the Humanities', 'HS', '2957 N Hoyne Ave', 'Chicago', 'IL', 60618, '(773) 534-5979', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610524.pdf', 'North-Northwest Side High School Network', 'NORTH-NORTHWEST SIDE COLLABORATIVE', 'NDA', 'Standard', 'Not on Probation', 'Not Enough Data', 'No', 'Strong', 70.0, 'NDA', 'NDA', 'Strong', 67.0, 'Average', 51.0, 'Weak', 'NDA', 'NDA', 'NDA', 'Strong', '57', 'Weak', '43', '92.70%', 7.1, '96.90%', '100.00%', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', None, None, None, None, 'NDA', 'NDA', 'NDA', 'NDA', '14.7', '13.7', 'NDA', '16', '1.3', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 232, 33, '87.6', 1161870.556, 1919857.44, 41.93576106, -87.68052441, 5, 'NORTH CENTER', 1, 19, '(41.93576106, -87.68052441)'),
 (18, 610334, 'Al Raby High School', 'HS', '3545 W Fulton Blvd', 'Chicago', 'IL', 60624, '(773) 534-6755', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610334.pdf', 'West Side High School Network', 'WEST SIDE COLLABORATIVE', 'No', 'Track_E', 'Probation', 'Level 3', 'No', 'Average', 52.0, 'NDA', 'NDA', 'Average', 51.0, 'Average', 53.0, 'Weak', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', '80.30%', 14.0, '95.10%', '100.00%', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', None, None, None, None, 'NDA', 'NDA', 'NDA', 'NDA', '12.8', '13', '14.1', '14', '1.2', '15.2', '1.1', '13.5', '72.2', '64.2', 551, 34, '74.3', 1152512.295, 1901720.85, 41.886183, -87.71539705, 27, 'EAST GARFIELD PARK', 28, 11, '(41.886183, -87.71539705)'),
 (40, 400018, 'Austin Business and Entrepreneurship Academy High School', 'HS', '231 N Pine Ave', 'Chicago', 'IL', 60644, '(773) 534-6316', None, 'West Side High School Network', 'WEST SIDE COLLABORATIVE', 'No', 'Non_Standard', 'Probation', 'Level 3', 'No', 'Weak', 38.0, 'NDA', 'NDA', 'Average', 41.0, 'Weak', 33.0, 'Weak', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', '76.00%', 12.4, '0.00%', '98.20%', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', None, None, None, None, 'NDA', 'NDA', 'NDA', 'NDA', '12', '12', '14.3', '12.9', '0.9', '14.4', '0.1', '12.2', '65.6', '51.3', 316, 36, 'NDA', 1139494.763, 1901274.258, 41.88520477, -87.763211

Copyright © 2020 This notebook and its source code are released under the terms of the [MIT License](https://bigdatauniversity.com/mit-license?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ).


## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>

## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                   |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
